File: OnlineStore/Api/CommonEntities/UserOrder.cs
using System;
using System.Collections.Generic;
using CommonEntities.Additional;

namespace CommonEntities
{
	public class UserOrder : BaseDbEntity
	{


		public string Address { get; set; }
		public DateTime DateOrder { get; set; }
		public int UserId { get; set; }
		public List<Product> Products { get; set; }



		public OrderStatus OrderStatus { get; set; }
		public UserSystem UserSystem { get; set; }


		#region Конструктор

		public UserOrder()
		{

		}
		public UserOrder(int orderId, DateTime dateOrder, string address, string orderStatus, int idUser, List<Product> products)
			: base(orderId)
		{
			Address = address;
			OrderStatus = new OrderStatus(orderStatus);
			DateOrder = dateOrder;
			UserId = idUser;
			Products = products;
			UserSystem = null;

		}

		public UserOrder( DateTime dateOrder, string address, string orderStatus, int idUser, List<Product> products)
		{
			Address = address;
			OrderStatus = new OrderStatus(orderStatus);
			DateOrder = dateOrder;
			UserId = idUser;
			Products = products;
			UserSystem = null;

		}

		public UserOrder(int orderId, DateTime dateOrder, string address, OrderStatus orderStatus, UserSystem userSystem, List<Product> products)
			: base(orderId)
		{
			DateOrder = dateOrder;
			Address = address;
			OrderStatus = orderStatus;
			UserSystem = userSystem;
			Products = products;
		}
		public UserOrder(int orderId, DateTime dateOrder, string address, string orderStatus, UserSystem userSystem, List<Product> products)
			: base(orderId)
		{
			DateOrder = dateOrder;
			Address = address;
			OrderStatus = new OrderStatus(orderStatus);
			UserSystem = userSystem;
			Products = products;
		}
		

		#endregion


	}
}

File: OnlineStore/OnlineStoreUI/src/model/entities/apiEntities/userOrder.ts
import { Product } from './product';
import { OrderStatus } from './additional/orderStatus';
import { SystemUserData } from './systemUserData';

export class UserOrder{
    idEntity: number;
    dateOrder:Date;
    address:string;
    products:Product[];
    orderStatus:OrderStatus;
    userSystemData: SystemUserData;

    constructor(idEntity: number, dateOrder:Date, address:string, products:Product[], orderStatus:OrderStatus, userSystemData: SystemUserData){
        this.idEntity = idEntity;
        this.dateOrder = dateOrder;
        this.address = address;
        this.products = products;
        this.orderStatus = orderStatus;
        this.userSystemData = userSystemData;
    }
}

